### DISCUSSION TITLE
Build-time style generation

### DISCUSSION THREAD

**Discussion by a7medev** [jaredh159/tailwind-react-native-classnames#147] [Ideas]
## The Problem

Generating styles at runtime can be a bit slow and somewhat unnecessary, since you have to parse the string passed to the function, generate the styles and cache them every time the user opens the app.

## Proposed Solution

**Generating styles at build-time** can improve the performance a lot since the styles will be generated once at build-time and the library will not need to re-generate it on every run, kinda like what styled-components does.

**Comment by jaredh159** [comment]
thanks for your input!

i'm wondering, do you have some real-world benchmarking measurements or some other data to show that this is _actually a problem_ in production? to be honest, i'm not terribly interested in tackling this without some hard data showing it's a performance problem for real. the lib is already heavily optimized, with many common utilities pre-parsed, and has an aggressive caching layer.

if you want something that generates a map of styles ahead of time `tailwind-rn` basically does that, but they don't support nearly as many of the device-context-aware features or the new arbitrary jit-style stuff that this library does, so it's a trade-off.
